nsq/writer.py

```diff
@@ -145,6 +149,7 @@ def _pub(self, command, topic, msg, delay_ms=None, callback=None):
             conn.send(cmd(*args))
         except Exception:
             logger.exception('[%s] failed to send %s' % (conn.id, command))
+            callback(None, protocol.SendError('sending error'))
```

Comment:
I think that without this, the caller is not informed that it possibly failed to send the message (even after seemingly having a connection).